storage/src/main/java/org/apache/uniffle/storage/handler/api/ShuffleDeleteHandler.java

```diff
@@ -24,5 +24,5 @@ public interface ShuffleDeleteHandler {
    *
    * @param appId ApplicationId for delete
    */
-  void delete(String[] storageBasePaths, String appId, String user);
+  void delete(String[] storageBasePaths, String appId, String user, String shuffleServerId);
```

Comment:
I don't like this makes the shuffleServerId populated in this method var. I hope this shuffleServerId as this handler's private member var

Reply:
If so, we don't need to modify the AppPurgeEvent's struction

Reply:
> If so, we don't need to modify the AppPurgeEvent's struction

I'm afraid not, we still need to modify `AppPurgeEvent`. Because we need to know whether the data is clear for expired

Reply:
> I don't like this makes the shuffleServerId populated in this method var. I hope this shuffleServerId as this handler's private member var

Done

Reply:
> I'm afraid not, we still need to modify AppPurgeEvent. Because we need to know whether the data is clear for expired

Why? From my side, this operation could be applied to all app purge, not only for expire app

Reply:
> Why? From my side, this operation could be applied to all app purge, not only for expire app

For performance. Deleting the application directory only requires one request.

Reply:
Got it. Make sense.

Reply:
Maybe we should add some comments to let more people know this.